sdk/core/azure-core/inc/azure/core/http/curl/curl connection_pool.hpp

```diff
@@ -27,6 +27,77 @@ namespace Azure { namespace Core { namespace Test {
 
 namespace Azure { namespace Core { namespace Http {
 
+  /**
+   * @brief The available options to set libcurl ssl options.
+   *
+   * @remark The SDK will map the enum option to libcurl's specific option. See more info here:
+   * https://curl.haxx.se/libcurl/c/CURLOPT_SSL_OPTIONS.html
+   *
+   */
+  enum class CurlTransportSSLOptions
```

Comment:
Should this enum be a bitmask/flag to match what curl expects? The user is allowed to set more than one of these.

Reply:
Can we use a flags enum which have bits set, like these?
https://github.com/Azure/azure-sdk-for-cpp/blob/3a4e1f3b491f37179df8eca56d8e136d9e8b909a/sdk/storage/azure-storage-common/inc/azure/storage/common/account_sas_builder.hpp#L40-L77

Reply:
Not really. What we need is a struct instead of the Enum which can hold each value.
Then when creating the connection, all enabled options needs to be merged to create the long value that we want to use.

Reply:
Why can't we have a bit flag? Why is a struct of bools better?

Reply:
I bit flag (just a long value) would require `the user` to know what's the number (bit) that they need to set for the option.
I want to make sure that only the valid number (the supported ones) can be set.
That's why I am offering this as part of the abstraction and I would do the bit flag for the user.